cli/azd/cmd/env.go

```diff
@@ -150,6 +168,309 @@ func (e *envSetAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	return nil, nil
 }
 
+func newEnvSetSecretFlags(cmd *cobra.Command, global *internal.GlobalCommandOptions) *envSetSecretFlags {
+	flags := &envSetSecretFlags{}
+	flags.Bind(cmd.Flags(), global)
+
+	return flags
+}
+
+type envSetSecretFlags struct {
+	internal.EnvFlag
+	global *internal.GlobalCommandOptions
+}
+
+func (f *envSetSecretFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	f.EnvFlag.Bind(local, global)
+	f.global = global
+}
+
+type envSetSecretAction struct {
+	console            input.Console
+	azdCtx             *azdcontext.AzdContext
+	env                *environment.Environment
+	envManager         environment.Manager
+	flags              *envSetFlags
+	args               []string
+	prompter           prompt.Prompter
+	kvService          keyvault.KeyVaultService
+	entraIdService     entraid.EntraIdService
+	subResolver        account.SubscriptionTenantResolver
+	userProfileService *azapi.UserProfileService
+}
+
+func (e *envSetSecretAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+
+	if len(e.args) < 1 {
+		return nil, fmt.Errorf(
+			"no <name> provided. Please provide a name as argument like: 'azd env set-secret <name>'")
+	}
+	secretName := e.args[0]
+
+	// When AZD_ENV_SET_SECRET_LIST_WITHOUT_NUMBERS is defined in azd env, azd will not add numbers to the list when
+	// asking to select options. For example, instead of showing "1. Option 1", it will show "Option 1". This is useful
+	// when the user wants to prefill the selection in stdin before calling azd env set-secret (e.g. in a script).
+	_, listWithoutNumbers := e.env.LookupEnv("AZD_ENV_SET_SECRET_LIST_WITHOUT_NUMBERS")
```

Comment:
Also, happy if we want to rely on `console.IsSpinnerInteractive()` (badly named -- it's simply a determination of `isTty`) to drop the numbers -- I know we do have precedence in other places of the codebase.